tensorboard/BUILD

```diff
@@ -23,9 +23,23 @@ py_binary(
     srcs_version = "PY2AND3",
     visibility = ["//visibility:public"],
     deps = [
+        "//tensorboard/compat:tensorflow",
         ":default",
         ":program",
-        "//tensorboard:expect_tensorflow_installed",
```

Comment:
For this PR, I think we want to be a bit more careful and still require a full TensorFlow installation for the main TensorBoard build.

The issue I'm concerned about is that as-is, this PR will negatively affect the regular, TF-dependent build of TensorFlow in that now if a user accidentally tries to run TensorBoard without a TensorFlow installation available, instead of getting an easy-to-interpret error that `import tensorflow as tf` failed, they'll instead get a much more confusing situation where TensorBoard partly works due to the wrapper (e.g. the no-TF-compatible plugins work) but any parts that still require real TensorFlow will fail with strange errors about the wrapper class lacking an attribute/module that the code expected.

We should make this scenario work better, but that requires a bit of thought about what the UX should be (e.g. do plugins self-report something like "install TensorFlow to use me" via the web GUI, do they error out on the console, do we require users to opt-into TensorFlow-less TensorBoard via a CLI flag like `--no-tf`, etc.).

For now, let's keep the status quo, and restrict the no-TF case to if you build `//tensorboard:tensorboard-notf` directly.  I think we can do this by changing the `compat/__init__.py` import logic to import tensorflow uncondtionally.  So the result would look like

```python
try:
  # Check if non-TensorFlow is enabled by build rules based on this import
  from . import notf  # noqa
  USING_TF = False
except ImportError:
  import tensorflow as tf
  USING_TF = True

if not USING_TF:
  from . import tensorflow_stub as tf  # noqa
```